web-console/src/druid-models/ingestion-spec/ingestion-spec.tsx

```diff
@@ -2429,20 +2429,31 @@ export function updateSchemaWithSample(
 
   let newSpec = spec;
 
-  if (dimensionMode === 'auto-detect') {
-    newSpec = deepSet(newSpec, 'spec.dataSchema.dimensionsSpec.useSchemaDiscovery', true);
-    newSpec = deepSet(newSpec, 'spec.dataSchema.dimensionsSpec.includeAllDimensions', true);
-    newSpec = deepSet(newSpec, 'spec.dataSchema.dimensionsSpec.dimensionExclusions', []);
-    newSpec = deepDelete(newSpec, 'spec.dataSchema.dimensionsSpec.dimensions');
-  } else {
-    newSpec = deepDelete(newSpec, 'spec.dataSchema.dimensionsSpec.useSchemaDiscovery');
-    newSpec = deepDelete(newSpec, 'spec.dataSchema.dimensionsSpec.includeAllDimensions');
-    newSpec = deepDelete(newSpec, 'spec.dataSchema.dimensionsSpec.dimensionExclusions');
-    newSpec = deepSet(
-      newSpec,
-      'spec.dataSchema.dimensionsSpec.dimensions',
-      getDimensionSpecs(sampleResponse, typeHints, guessNumericStringsAsNumbers, rollup),
-    );
+  switch (schemaMode) {
+    case 'flexible':
+      newSpec = deepSet(newSpec, 'spec.dataSchema.dimensionsSpec.useSchemaDiscovery', true);
+      newSpec = deepSet(newSpec, 'spec.dataSchema.dimensionsSpec.includeAllDimensions', true);
```

Comment:
setting `useSchemaDiscovery` includes the functionality of `includeAllDimensions`, so it isn't necessary to set to support partial schema definition